protocol/messenger_emoji_reactions.go

```diff
@@ -31,6 +35,27 @@ func ConvertEmojiIDToString(emojiID protobuf.EmojiReaction_Type) string {
 
 }
 
+func (m *Messenger) CheckMaxNumberOfEmojiReactionsPerMessage(chatID string, messageID string, emoji string) error {
+	// Limit the amount of emoji reactions per message to avoid spam
+	numberOfEmojis, err := m.persistence.EmojiReactionCountByChatIDMessageID(chatID, messageID)
```

Comment:
Maybe we have the message cached, and there's no need to query the DB?

Reply:
We don't have a cache for messages or reactions. Also `common.Message` doesn't contain the reactions. We need to query them from the DB separately.